lib/HttpClient.js

```diff
@@ -414,17 +446,32 @@ function HttpClient(options) {
     this.socketPath = options.socketPath || false;
     this.url = options.url ? url.parse(options.url) : {};
 
-    if (process.env.https_proxy) {
-        this.proxy = url.parse(process.env.https_proxy);
-    } else if (process.env.http_proxy) {
-        this.proxy = url.parse(process.env.http_proxy);
+    // HTTP proxy: `options.proxy` wins, else `https_proxy`/`http_proxy` envvars
+    // (upper and lowercase) are used.
+    if (options.proxy === false) {
+        this.proxy = false;
```

Comment:
We tend to use the convention of `self` instead of `this` when accessing or mutating a variable that has already been declared. Mind sticking to this convention?